build.psm1

```diff
@@ -398,81 +437,224 @@ function Start-BuildNativeUnixBinaries {
+            <licenseUrl>https://github.com/PowerShell/PowerShell/blob/master/LICENSE.txt</licenseUrl>
+            <tags>PowerShell</tags>
+            <language>en-US</language>
+            <copyright>Copyright (c) Microsoft Corporation. All rights reserved.</copyright>
+            <contentFiles>
+                <files include="**/*" buildAction="None" copyToOutput="true" flatten="false" />
+            </contentFiles>
+        </metadata>
 </package>
 '@
 
-    $plugingNuspec -f $PluginVersion | Out-File -FilePath (Join-Path $plugingPackageRoot -ChildPath 'psrp.windows.nuspec') -Force
+    $Nuspec -f $Version | Out-File -FilePath (Join-Path $PackageRoot -ChildPath 'Microsoft.PowerShell.Native.nuspec') -Force
+
+    Write-Verbose "Listing all the files under PackageRoot" -Verbose
+    Get-ChildItem $PackageRoot -Recurse | ForEach-Object { Write-Verbose $_.FullName -Verbose}
+
+    if(-not (Test-Path $NuGetOutputPath))
+    {
+        $null = New-Item $NuGetOutputPath -Force -Verbose -ItemType Directory
+    }
 
     try {
-        Push-Location $plugingPackageRoot
-        nuget.exe pack .
+        Push-Location $PackageRoot
+        nuget.exe pack . -OutputDirectory $NuGetOutputPath
     }
     finally {
         Pop-Location
     }
+
+    if(-not $SkipCleanup -and (Test-Path $tempExtractionPath))
+    {
+        Remove-Item $tempExtractionPath -Recurse -Force -ErrorAction SilentlyContinue
+    }
 }
 
-function BuildPwrshPlugingNugetLayout
+function PlaceUnixBinaries
 {
+    [CmdletBinding()]
     param(
-        [string] $Destination,
-        [string] $Arch
+        [Parameter(Mandatory = $true)]
+        $PackageRoot,
+
+        [Parameter(Mandatory = $true)]
+        [ValidateScript({Test-Path $_ -PathType Container})]
+        $BinFolderLinux,
+
+        [Parameter(Mandatory = $true)]
+        [ValidateScript({Test-Path $_ -PathType Container})]
+        $BinFolderLinuxARM,
+
+        [Parameter(Mandatory = $true)]
+        [ValidateScript({Test-Path $_ -PathType Container})]
+        $BinFolderOSX
     )
 
-    try {
-        Start-BuildNativeWindowsBinaries -Configuration 'Release' -Arch $Arch -Clean -ErrorAction Stop
-    }
-    catch {
-        throw "Build failed: $_.Message"
+    $RuntimePathLinux = New-Item -ItemType Directory -Path (Join-Path $PackageRoot -ChildPath 'runtimes/linux-x64/native') -Force
+    $RuntimePathLinuxARM = New-Item -ItemType Directory -Path (Join-Path $PackageRoot -ChildPath 'runtimes/linux-arm/native') -Force
+    $RuntimePathOSX = New-Item -ItemType Directory -Path (Join-Path $PackageRoot -ChildPath 'runtimes/osx/native') -Force
+
+    Copy-Item "$BinFolderLinux\*" -Destination $RuntimePathLinux -Verbose
+    Copy-Item "$BinFolderLinuxARM\*" -Destination $RuntimePathLinuxARM -Verbose
+    Copy-Item "$BinFolderOSX\*" -Destination $RuntimePathOSX -Verbose
+}
+
+function PlaceWindowsNativeBinaries
```

Comment:
you should add some form of header... at least a synopsis

Reply:
Added a synopsis for all new functions